test/compress/functions.js

```diff
@@ -2031,6 +2031,29 @@ inline_true: {
     ]
 }
 
+issue_2842: {
+    options = {
+        inline: true,
+        toplevel: true,
+        reduce_vars: true,
+        reduce_funcs: true,
+    }
+    input: {
+        {
+            const data = function (data) {
+                return data[data[0]];
+            }([1, 2, 3]);
+        }
```

Comment:
@fabiosantoscode If you put the entire standalone test case from https://github.com/mishoo/UglifyJS2/issues/2842#issuecomment-359587006 here and add an `expect_stdout` clause to this test then it can be automatically retested with a [variety of minify options](https://github.com/mishoo/UglifyJS2/blob/harmony/test/ufuzz.json) with reminify.

Reply:
You may have to rebase from latest harmony to have [reminify](https://github.com/mishoo/UglifyJS2/blob/569757d14dbbe04f66af65d7bf0600b3a75efd96/test/run-tests.js#L211) work again.

Reply:
I've done the suggested changes now :)